main.go

```diff
@@ -0,0 +1,227 @@
+		}
+		artifacts.Packet = &packet
+
+		packetImage := StreamCloudImage{Image: "fedora_coreos_stable"}
+		cloudImage.Packet = &packetImage
+	}
+
+	if releaseArch.Media.Openstack != nil {
+		openstack := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Openstack.Artifacts,
+		}
+		artifacts.Openstack = &openstack
+	}
+
+	if releaseArch.Media.Qemu != nil {
+		qemu := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Qemu.Artifacts,
+		}
+		artifacts.Qemu = &qemu
+	}
+
+	if releaseArch.Media.Virtualbox != nil {
+		virtualbox := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Virtualbox.Artifacts,
+		}
+		artifacts.Virtualbox = &virtualbox
+	}
+
+	if releaseArch.Media.Vmware != nil {
+		vmware := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Vmware.Artifacts,
+		}
+		artifacts.Vmware = &vmware
+	}
+
+	if releaseArch.Media.Metal != nil {
+		metal := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Metal.Artifacts,
+		}
+		artifacts.Metal = &metal
+	}
+
+	streamArch := StreamArch{
+		Artifacts: artifacts,
+		Updates:   StreamUpdates{Release: release.Release},
+		Images:    &cloudImage,
+	}
+	return streamArch
+}
+
+func main() {
+	var releasesURL string
+
+	flag.StringVar(&releasesURL, "release-index", "", "Release index location for the required stream")
+	var overrideFilename string
+	flag.StringVar(&overrideFilename, "override", "", "Override file location for the required stream")
+
+	flag.Parse()
+
+	if releasesURL == "" {
+		fmt.Println("Please specify release index url")
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	releasePath, err := ReleaseURL(releasesURL)
+
+	if err != nil {
+		fmt.Printf("Error with Release Index %v", err)
+		os.Exit(1)
+	}
+
+	var release Release
+	resp, err := http.Get(releasePath)
+	if err != nil {
+		fmt.Printf("Error while fetching %v", err)
+		os.Exit(1)
+	}
+
+	defer resp.Body.Close()
+
+	decoder := json.NewDecoder(resp.Body)
+	if err := decoder.Decode(&release); err != nil {
+		fmt.Printf("Error while decoding json %v", err)
+		os.Exit(1)
+	}
+
+	arch := StreamArchitectures{}
+	if release.Architectures.X8664 != nil {
+		x8664 := releaseToStream(release.Architectures.X8664, release)
+		arch.X8664 = &x8664
+	}
+
+	if release.Architectures.PowerPC != nil {
+		powerPC := releaseToStream(release.Architectures.PowerPC, release)
+		arch.PowerPC = &powerPC
+	}
+
+	if release.Architectures.Arm64 != nil {
+		arm64 := releaseToStream(release.Architectures.Arm64, release)
+		arch.Arm64 = &arm64
+	}
+
+	streamMetadata := StreamMetadata{
+		Stream:        release.Stream,
+		Metadata:      Metadata{LastModified: time.Now().UTC().Format(time.RFC3339)},
+		Architectures: arch,
+	}
+
+	file, err := os.Create("stream.json")
+	if err != nil {
+		fmt.Printf("Can't Open file: %v", err)
+		os.Exit(1)
+	}
+
+	defer file.Close()
+
+	if overrideFilename != "" {
```

Comment:
Is it intended that `stream.json` is still produced if `overrideFilename` is specified? I would expect that if specified the output would be routed to that location instead of `stream.json`.

Reply:
As far as I understand yes, stream.json will be automatically generated in both cases - a) when release.json is updated  b) when a override file is specified

Reply:
Ah, I misread this initially as an additional output file. Don't mind me.